### PR TITLE
Feature/quack 5528 fragment java

### PR DESCRIPTION
**Pull request by brandon-sensibill**

### CODE REVIEW

File: app/src/main/java/com/getsensibill/spendmanager/demo/fragment/FragmentJavaActivity.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        binding = ActivityFragmentJavaBinding.inflate(getLayoutInflater());
+        setContentView(binding.getRoot());
+
+        if(savedInstanceState == null) {
+            WebUiFragment fragment = new WebUiFragment();
+            Bundle bundle = fragment.getArguments();
+            if (bundle == null) {
```
Comment (patshu) [inline app/src/main/java/com/getsensibill/spendmanager/demo/fragment/FragmentJavaActivity.java]: If we're always creating a new fragment above, `.getArguments()` will always be null, so we can just say `Bundle bundle = new Bundle();`

File: app/src/main/java/com/getsensibill/spendmanager/demo/fragment/FragmentJavaActivity.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+    /**
+     * Required Override when using [WebUiFragment.Listener]
+     *
+     * Handle when the network error fragment needs to be navigated to. We have provided a
+     * [WebUiNetworkErrorFragment] that you can display to the user. Or you can use your
+     * own fragment or even an alternative logic flow if desired.
+     */
+    @Override
+    public void onDisplayNetworkError(boolean b) {
```
Comment (patshu) [inline app/src/main/java/com/getsensibill/spendmanager/demo/fragment/FragmentJavaActivity.java]: Did we lose the parameter name here?

File: app/src/main/java/com/getsensibill/spendmanager/demo/fragment/FragmentJavaActivity.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        WebUiNetworkErrorFragment fragment = new WebUiNetworkErrorFragment();
+        Bundle bundle = fragment.getArguments();
+        if (bundle == null) {
```
Comment (patshu) [inline app/src/main/java/com/getsensibill/spendmanager/demo/fragment/FragmentJavaActivity.java]: Same comment as above

File: app/src/main/java/com/getsensibill/spendmanager/demo/fragment/FragmentJavaActivity.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+                    NavigationIntent.DASHBOARD.INSTANCE
+            );
+
+            // Pass in a custom Theme override. Defaults to null
+            ProgrammaticTheme theme = new ProgrammaticTheme(new Brand());
+            bundle.putParcelable(WebUiFragment.ARG_PROGRAMMATIC_THEME_OVERRIDE, theme);
+
+            fragment.setArguments(bundle);
+
+            FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
```
Comment (patshu) [inline app/src/main/java/com/getsensibill/spendmanager/demo/fragment/FragmentJavaActivity.java]: Not a big thing, but i think `FragmentManager.Transaction`s are builder style, shouldn't need to give it a variable, can probably just chain commands

File: app/src/main/java/com/getsensibill/spendmanager/demo/fragment/FragmentJavaActivity.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+            transaction.replace(R.id.fragment_container, fragment, "FRAGMENT_TAG");
```
Comment (patshu) [inline app/src/main/java/com/getsensibill/spendmanager/demo/fragment/FragmentJavaActivity.java]: Small inconsistency here (don't need to change if you don't want to), in the `Kotlin` implementation this was `binding.fragmentContainer.id`, but here we're using `R`

File: app/src/main/java/com/getsensibill/spendmanager/demo/fragment/FragmentJavaActivity.java (hunk 2 above)
Comment (brandon-sensibill) [reply app/src/main/java/com/getsensibill/spendmanager/demo/fragment/FragmentJavaActivity.java]: ya weird. when I use the intellescence to make it, it comes out like that. Updated.

File: app/src/main/java/com/getsensibill/spendmanager/demo/fragment/FragmentJavaActivity.java (hunk 1 above)
Comment (brandon-sensibill) [reply app/src/main/java/com/getsensibill/spendmanager/demo/fragment/FragmentJavaActivity.java]: updated

File: app/src/main/java/com/getsensibill/spendmanager/demo/fragment/FragmentJavaActivity.java (hunk 3 above)
Comment (brandon-sensibill) [reply app/src/main/java/com/getsensibill/spendmanager/demo/fragment/FragmentJavaActivity.java]: updated

File: app/src/main/java/com/getsensibill/spendmanager/demo/fragment/FragmentJavaActivity.java (hunk 4 above)
Comment (brandon-sensibill) [reply app/src/main/java/com/getsensibill/spendmanager/demo/fragment/FragmentJavaActivity.java]: updated

File: app/src/main/java/com/getsensibill/spendmanager/demo/fragment/FragmentJavaActivity.java (hunk 5 above)
Comment (brandon-sensibill) [reply app/src/main/java/com/getsensibill/spendmanager/demo/fragment/FragmentJavaActivity.java]: Updated for consistency

**Review by patshu — APPROVED** [review]

**Pull request merged**